src/buildThunks.ts

```diff
@@ -42,7 +42,7 @@ export function buildThunks<InternalQueryArgs, ReducerPath extends string>({
     `${reducerPath}/executeQuery`,
     (arg, { signal, rejectWithValue }) => {
       return baseQuery(arg.internalQueryArgs, { signal, rejectWithValue }).then(
```

Comment:
On a general note: we should not use `.then` here, but change the whole method to `async` and `await` the result.
In the case the `baseQuery` would return synchronously, this would otherwise crash with "no property `.then` on ...".
While you're already here, can you do that?